packages/code-studio/src/main/AppMainContainer.tsx

```diff
@@ -123,7 +123,7 @@ interface AppMainContainerProps {
   match: {
     params: { notebookPath: string };
   };
-  connection: IdeConnection;
+  connection?: IdeConnection;
```

Comment:
Is this true? I can see `session` being `undefined`, but in what case is `connection` ever `undefined`? This shouldn't be a case we have to handle.

Reply:
Ah true. Should update that type in redux as well then.

Reply:
I think it is already up to date? `dashboardData: Record<string, DashboardData>;` can also be an empty record.

Reply:
Ah whoops I was looking at old code, looks good.